compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -314,6 +314,10 @@ impl<'a> Linker for GccLinker<'a> {
                 self.cmd.arg("-static");
                 self.build_dylib(out_filename);
             }
+            LinkOutputKind::WasiReactorExe => {
+                self.cmd.arg("--entry");
```

Comment:
I'm not sure that `--entry` is a valid argument to clang itself, but what you can do is `self.linker_arg` which automatically prepends the `-Wl,` I believe